File: 3043-find-the-length-of-the-longest-common-prefix/3043-find-the-length-of-the-longest-common-prefix.cpp
class Solution {
public:
    int longestCommonPrefix(vector<int>& arr1, vector<int>& arr2) {
        int res = 0;
        unordered_set<int> pres;
        for(int i = 0; i < arr1.size(); i++)
        {
            int a = arr1[i];
            while(!pres.count(a) && a > 0)
            {
                pres.insert(a);
                a /= 10;
            }
        }
        for(int i = 0; i < arr2.size(); i++)
        {
            int b = arr2[i];
            while(b > 0)
            {
                if(pres.count(b))
                {
                    res = max(res, (int)to_string(b).length());
                    break;
                }
                b /= 10;
            }
        }
        return res;
    }
};

File: 3043-find-the-length-of-the-longest-common-prefix/3043-find-the-length-of-the-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, arr1: List[int], arr2: List[int]) -> int:
        res = 0
        pres = set()
        for a in arr1:
            num = str(a)
            for i in range(len(num)):
                pres.add(num[:i + 1])
        for b in arr2:
            num = str(b)
            for i in range(len(num)):
                if num[:i + 1] in pres:
                    res = max(res, i + 1)
        return res